hi-ml-histopathology/src/histopathology/utils/report_utils.py

```diff
@@ -0,0 +1,339 @@
+    :param output_filename: Filename of the output CSVs to download.
+    :return: A dictionary of dataframes with the sorted cross-validation indices as keys.
+    """
+    parent_run = get_aml_run_from_run_id(parent_run_id, aml_workspace)
+
+    all_outputs_dfs = {}
+    for child_run in parent_run.get_children():
+        child_run_index = get_tags_from_hyperdrive_run(child_run, crossval_arg_name)
+        if child_run_index is None:
+            raise ValueError(f"Child run expected to have the tag '{crossval_arg_name}'")
+        child_dir = download_dir / str(child_run_index)
+        try:
+            child_csv = download_from_run_if_necessary(child_run,
+                                                       remote_dir=Path("outputs"),
+                                                       download_dir=child_dir,
+                                                       filename=output_filename)
+            all_outputs_dfs[child_run_index] = pd.read_csv(child_csv)
+        except Exception as e:
+            print(f"Failed to download {output_filename} for run {child_run.id}: {e}")
+    return dict(sorted(all_outputs_dfs.items()))
+
+
+def collect_crossval_metrics(parent_run_id: str, download_dir: Path, aml_workspace: Workspace,
+                             crossval_arg_name: str = "cross_validation_split_index") -> pd.DataFrame:
+    """Fetch metrics logged to Azure ML from cross-validation runs as a dataframe.
+
+    Will only download the metrics if they do not already exist locally, as this can take several
+    seconds for each child run.
+
+    :param parent_run_id: Azure ML run ID for the parent Hyperdrive run.
+    :param download_dir: Directory where to save the downloaded metrics as `aml_metrics.pickle`.
+    :param aml_workspace: Azure ML workspace in which the runs were executed.
+    :param crossval_arg_name: Name of the Hyperdrive argument used for indexing the child runs.
+    :return: A dataframe in the format returned by :py:func:`~health_azure.aggregate_hyperdrive_metrics()`.
+    """
+    # Save metrics as a pickle because complex dataframe structure is lost in CSV
+    metrics_pickle = download_dir / "aml_metrics.pickle"
+    if metrics_pickle.is_file():
+        print(f"AML metrics file already exists at {metrics_pickle}")
+        with open(metrics_pickle, 'rb') as f:
+            metrics_df = pickle.load(f)
+    else:
+        metrics_df = aggregate_hyperdrive_metrics(run_id=parent_run_id,
+                                                  child_run_arg_name=crossval_arg_name,
+                                                  aml_workspace=aml_workspace)
+        metrics_pickle.parent.mkdir(parents=True, exist_ok=True)
+        print(f"Writing AML metrics file to {metrics_pickle}")
+        with open(metrics_pickle, 'wb') as f:
+            pickle.dump(metrics_df, f)
+    return metrics_df.sort_index(axis='columns')
+
+
+def plot_roc_curve(labels: Sequence, scores: Sequence, label: str, ax: Axes) -> None:
```

Comment:
Would it make sense to put these plotting functions in `histopathology.utils.analysis_plot_utils`?

Reply:
Also line 254-308

Reply:
All now migrated to `analysis_plot_utils`.